src/components/app.jsx

```diff
@@ -29,9 +33,21 @@ export default class App extends React.Component {
     }
   }
 
+  saveCards() {
+    const cards = JSON.stringify(this.state.cards);
+    localStorage.setItem('flash-cards', cards)
+  }
+
+  addCards(card) {
+    const cards = this.state.cards.slice(0);
+    const newCards = cards.concat(card);
```

Comment:
Concat doesn't modify the original array, it returns a copy. So you can skip the slicing.